### PR TITLE
Created note listing screen.

### LINKED ISSUE #5: Create note listing page UI

### PR DESCRIPTION
**Pull request by ubaidchalil**
Fixes #5 

- Added notes dummy data
- Added design for displaying note items
- Fixed a few labels 
- Added side menu bar for note list screen

### CODE REVIEW

File: app/javascript/src/components/Common/SideMenuBar.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { Typography } from "neetoui";
+import { MenuBar } from "neetoui/layouts";
+import PropTypes from "prop-types";
+
+const SideMenuBar = ({ showSideMenuBar, title, menuData }) => {
+  const [isSearchCollapsed, setIsSearchCollapsed] = useState(false);
+
+  return (
+    <MenuBar showMenu={showSideMenuBar} title={title}>
+      {menuData.filterTypes?.map(({ label, dataCount, active }, idx) => (
```
Comment (yedhink) [inline app/javascript/src/components/Common/SideMenuBar.jsx:13]: Why's there an option chain over here?

File: app/javascript/src/components/Common/SideMenuBar.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      ))}
+    </MenuBar>
+  );
+};
+
+SideMenuBar.propTypes = {
+  showSideMenuBar: PropTypes.bool,
+  title: PropTypes.string,
+  menuData: PropTypes.object,
+};
```
Comment (yedhink) [inline app/javascript/src/components/Common/SideMenuBar.jsx:79]: No need to add prop types unless the component is reused at more than one place. No need to remove any prop types if already added.

File: app/javascript/src/components/Dashboard/Notes/Card.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import React from "react";
+
+import { MenuVertical, Clock } from "neetoicons";
+import { Typography, Dropdown, Tag, Tooltip, Avatar } from "neetoui";
+import PropTypes from "prop-types";
+
+import {
+  formatDateToDayAndTime,
+  formatDateToTimeSinceFromNow,
+} from "utils/index";
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Card.jsx:10]: No need to mentioned `/index` when importing from `index` file. Webpack will auto resolve it.

File: app/javascript/src/components/Dashboard/Notes/Card.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+}) => (
+  <div className="my-2 w-full rounded-sm border border-solid p-4 shadow">
+    <div className="flex items-center justify-between">
+      <Typography style="h4" weight="semibold">
+        {title}
+      </Typography>
+      <Dropdown buttonStyle="text" icon={MenuVertical}>
+        <Menu>
+          <MenuItem.Button>Edit</MenuItem.Button>
+          <MenuItem.Button style="danger" onClick={() => alert("Hi")}>
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Card.jsx:32]: Update with a more sensible message other than `Hi`.

File: app/javascript/src/components/Dashboard/Notes/Card.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+          <Typography className="mx-2" style="body2">
+            {`${NOTES_STATUS_TITLE[status]} ${formatDateToTimeSinceFromNow(
+              updatedAt
+            )}`}
+          </Typography>
+        </Tooltip>
+        <Avatar
+          user={{
+            imageUrl: assignedContact?.profile_image_url,
+            name: assignedContact?.name,
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Card.jsx:61]: https://academy.bigbinary.com/learn-react/react-best-practices/avoid-overusing-optional-chaining

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 6]
```diff
@@ -1,107 +1,66 @@
...
-import { Delete } from "neetoicons";
-import { Button, PageLoader } from "neetoui";
-import { Container, Header, SubHeader } from "neetoui/layouts";
+import { Button } from "neetoui";
+import { Container, Header } from "neetoui/layouts";
 
-import notesApi from "apis/notes";
 import EmptyState from "components/Common/EmptyState";
+import SideMenuBar from "components/Common/SideMenuBar";
+import { notes as noteList } from "data";
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:9]: I don't understand the benefit of this renaming. Can you please explain to me the need for doing the same? Thanks.

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 7]
```diff
@@ -1,107 +1,66 @@
...
-          setSelectedNoteIds={setSelectedNoteIds}
-          onClose={() => setShowDeleteAlert(false)}
+      <Container>
+        <Header
+          title="All Notes"
+          actionBlock={
+            <Button icon="ri-add-line" label="Add note" size="small" />
+          }
+          menuBarToggle={() =>
+            setShowSideMenuBar(_showSideMenuBar => !_showSideMenuBar)
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:37]: Why have you prefixed with an `_`(underscore)?

File: app/javascript/src/utils/index.js [hunk 8]
```diff
@@ -1,3 +1,9 @@
+import dayjs from "dayjs";
 import * as R from "ramda";
 
 export const isPresent = R.pipe(R.either(R.isNil, R.isEmpty), R.not);
+
+export const formatDateToDayAndTime = date =>
+  dayjs(date).format("MMMM, hh:mmA");
+
+export const formatDateToTimeSinceFromNow = date => dayjs(date).fromNow();
```
Comment (yedhink) [inline app/javascript/src/utils/index.js:9]:
```suggestion
export const formatDateToTimeSinceNow = date => dayjs(date).fromNow();
```

**Review by yedhink — COMMENTED** [review]
@ubaidchalil Few comments added. I won't be pinpointing each and every place in your codebase, where similar issues ought to be fixed. That's your responsibility. If you have any queries, then make sure to clarify.

File: app/javascript/src/components/Dashboard/Notes/index.jsx (hunk 6 above)
Comment (ubaidchalil) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:9]: In the same file already declared a state as notes. That is the reason for renaming the notes as noteList which is imported from data.

File: app/javascript/src/components/Common/SideMenuBar.jsx (hunk 1 above)
Comment (ubaidchalil) [reply app/javascript/src/components/Common/SideMenuBar.jsx:13]: Yes. We don't need to add this check here.

**Comment by ubaidchalil** [comment]
@yedhink I added fixes for the above-mentioned issues and created a [separate PR](https://github.com/ubaidchalil/neetoui-challenge/pull/15) for it.

File: app/javascript/src/components/Dashboard/Notes/index.jsx (hunk 6 above)
Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:9]: @ubaidchalil Is `notes` imported from `data` a literal constant? If yes, why you have you named it in camelCase?

Comment (ubaidchalil) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:9]: Yes. But in a real scenario, the data would be in camelCase. I thought like that. 
 
 I will update it to the capital snake_case.

**Pull request merged**